go/bucketpool/bucketpool.go

```diff
@@ -0,0 +1,81 @@
+package bucketpool
+
+import (
+	"sync"
+)
+
+type sizedPool struct {
+	size int
+	pool sync.Pool
+}
+
+func newSizedPool(size int) *sizedPool {
+	return &sizedPool{
+		size: size,
+		pool: sync.Pool{
+			New: func() interface{} { return makeSlicePointer(size) },
```

Comment:
I benchmarked and there is no difference between `New` and `if i == nil` way now.

Reply:
there should be at least 1 alloc/op due to the interface{} framing. probably irrelevant for the simplification.